lib/restforce/db/worker.rb

```diff
@@ -126,65 +127,16 @@ def runner
         @runner ||= Runner.new(@options.fetch(:delay) { DEFAULT_DELAY })
       end
 
-      # Internal: Propagate unsynchronized records between the two systems for
-      # the passed mapping.
-      #
-      # mapping - A Restforce::DB::Mapping.
-      #
-      # Returns nothing.
-      def propagate(mapping)
-        Initializer.new(mapping, runner).run
-      end
-
-      # Internal: Remove synchronized records from the database when the
-      # Salesforce record no longer meets the mapping's conditions.
-      #
-      # mapping - A Restforce::DB::Mapping.
-      #
-      # Returns nothing.
-      def clean(mapping)
-        Cleaner.new(mapping, runner).run
-      end
-
-      # Internal: Collect a list of changes from recently-updated records for
-      # the passed mapping.
-      #
-      # mapping - A Restforce::DB::Mapping.
-      #
-      # Returns nothing.
-      def collect(mapping)
-        Collector.new(mapping, runner).run(@changes)
-      end
-
-      # Internal: Update the associated records and Salesforce lookups for
-      # records belonging to the passed mapping.
-      #
-      # mapping - A Restforce::DB::Mapping.
-      #
-      # Returns nothing.
-      def associate(mapping)
-        Associator.new(mapping, runner).run
-      end
-
-      # Internal: Apply the aggregated changes to the objects in both systems,
-      # according to the defined mappings.
-      #
-      # mapping - A Restforce::DB::Mapping.
-      #
-      # Returns nothing.
-      def synchronize(mapping)
-        Synchronizer.new(mapping).run(@changes)
-      end
-
       # Internal: Log a description and response time for a specific named task.
       #
-      # name    - A String task name.
-      # mapping - A Restforce::DB::Mapping.
+      # name       - A String task name.
+      # task_class - A Restforce::DB::Task subclass.
+      # mapping    - A Restforce::DB::Mapping.
       #
       # Returns a Boolean.
-      def task(name, mapping)
+      def run(name, task_class, mapping)
         log "  #{name} between #{mapping.database_model.name} and #{mapping.salesforce_model}"
-        runtime = Benchmark.realtime { yield }
+        runtime = Benchmark.realtime { task task_class, mapping }
```

Comment:
I kept it around as a separate method so I could invoke it in the tests (see: https://github.com/tablexi/restforce-db/blob/prevent_out_of_band_upserted_record_propagation/test/lib/restforce/db/worker_test.rb#L23).